packages/theme/src/cli/utilities/theme-command.ts

```diff
@@ -218,30 +215,57 @@ export default abstract class ThemeCommand extends Command {
    * Run the command in each valid environment concurrently
    * @param validEnvironments - The valid environments to run the command in
    */
-  private async runConcurrent(
-    validEnvironments: {environment: EnvironmentName; flags: FlagValues; session: AdminSession}[],
-  ) {
+  private async runConcurrent(validEnvironments: {environment: EnvironmentName; flags: FlagValues}[]) {
     const abortController = new AbortController()
 
-    await renderConcurrent({
-      processes: validEnvironments.map(({environment, flags, session}) => ({
-        prefix: environment,
-        action: async (stdout: Writable, stderr: Writable, _signal) => {
-          try {
-            await this.command(flags, session, true, {stdout, stderr})
-
-            // eslint-disable-next-line no-catch-all/no-catch-all
-          } catch (error) {
-            if (error instanceof Error) {
-              error.message = `Environment ${environment} failed: \n\n${error.message}`
-              renderError({body: [error.message]})
+    const stores = validEnvironments.map((env) => env.flags.store as string)
+    const uniqueStores = new Set(stores)
+    const runGroups =
+      stores.length === uniqueStores.size ? [validEnvironments] : this.createSequentialGroups(validEnvironments)
+
+    for (const runGroup of runGroups) {
+      // eslint-disable-next-line no-await-in-loop
+      await renderConcurrent({
+        processes: runGroup.map(({environment, flags}) => ({
+          prefix: environment,
+          action: async (stdout: Writable, stderr: Writable, _signal) => {
+            try {
+              const store = flags.store as string
+              await useThemeStoreContext(store, async () => {
+                const session = await this.createSession(flags)
+                await this.command(flags, session, true, {stdout, stderr})
+              })
+
+              // eslint-disable-next-line no-catch-all/no-catch-all
+            } catch (error) {
+              if (error instanceof Error) {
+                error.message = `Environment ${environment} failed: \n\n${error.message}`
+                renderError({body: [error.message]})
+              }
             }
-          }
-        },
-      })),
-      abortSignal: abortController.signal,
-      showTimestamps: true,
+          },
+        })),
+        abortSignal: abortController.signal,
+        showTimestamps: true,
+      })
+    }
+  }
+
+  /**
+   * Create groups of environments with unique flags.store values to run sequentially
+   * to prevent conflicts between environments acting on the same store
+   * @param environments - The environments to group
+   * @returns The environment groups
+   */
+  private createSequentialGroups(environments: {environment: EnvironmentName; flags: FlagValues}[]) {
```

Comment:
I assume running a command in multiple environments with the same store URL would be pretty rare, but I could picture something like `shopify theme info -e development -e production` where allowing it would be helpful